msal/tests/Test.MSAL.NET.Unit.net45/Integration/ApplicationGrantIntegrationTest.cs

```diff
@@ -0,0 +1,74 @@
+﻿using Microsoft.Identity.Client;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Test.Microsoft.Identity.LabInfrastructure;
+
+namespace Test.MSAL.NET.Unit.net45.Integration
+{
+    [TestClass]
+    public class ApplicationGrantIntegrationTest
+    {
+        public const string Authority = "";
+        public const string ClientId = "";
+        public const string RedirectUri = "http://localhost";
+        public string[] MsalScopes = { "https://graph.microsoft.com/.default" };
+        private string Password = "";
+
+        static ApplicationGrantIntegrationTest()
+        {
+            ILabService _labService = new LabServiceApi(new KeyVaultSecretsProvider());
+            // init password here - read from lab api
+        }
+
+        [TestMethod]
+        [TestCategory("ApplicationGrant_IntegrationTests")]
+        [Ignore]
+        public async Task ApplicationGrantIntegrationTestAsync()
+        {
+            var appCache = new TokenCache();
+            var userCache = new TokenCache();
+
+            var confidentialClient = new ConfidentialClientApplication(ClientId, Authority, RedirectUri,
+                new ClientCredential(Password), userCache, appCache);
+
+            var res = await confidentialClient.AcquireTokenForClientAsync(MsalScopes).ConfigureAwait(false);
```

Comment:
Most of our tests have this as `AuthenticationResult` instead of `var` maybe write out `result`...just makes it easier to understand quickly.

Reply:
VisualStudio suggests using var when you use the actual type and suggests using the actual type when using var :). I don't think there is a clear rule around the use of var, I typically use var only if the actual type is obvious.